archive_extractor/zip_archiver.go

```diff
@@ -104,6 +107,10 @@ func initZipReader(r io.ReaderAt, size int64) (*zip.Reader, error) {
 		}
 		n := 0
 		for {
+			zi := bytes.Index(buf[n:len], []byte(zoneInfoFileHeaderSignatureString))
```

Comment:
let's extract this check to a separate method with a meaningful name.
I think it worth to also add a comment with an explanation.